### PR TITLE
[Exporter Refactor] Fixing conv integer node deletion & sorting in core ops

### PR DESCRIPTION
**Pull request by corey-nm**
Title - matches old behavior now

### CODE REVIEW

File: src/sparseml/exporters/transforms/gemm_to_matmulinteger_add_cast_mul.py [hunk 1]
```diff
@@ -104,6 +104,8 @@ def transform(self, model: ModelProto) -> ModelProto:
 
             self._do_transform(model, match, attr)
 
+        graph = ONNXGraph(model)
+        graph.sort_nodes_topologically()
```
Comment (dbogunowicz) [inline src/sparseml/exporters/transforms/gemm_to_matmulinteger_add_cast_mul.py:108]: Why is this necessary here?

File: src/sparseml/exporters/transforms/convert_quantizable_matmul.py [hunk 2]
```diff
@@ -158,4 +157,5 @@ def transform(self, model: ModelProto) -> ModelProto:
                 f"Converted {count_converted_nodes} quantizable MatMul ops "
                 "to QLinearMatMul"
             )
+        ONNXGraph(model).sort_nodes_topologically()
```
Comment (dbogunowicz) [inline src/sparseml/exporters/transforms/convert_quantizable_matmul.py:160]: Why is this necessary here?

**Review by dbogunowicz — APPROVED** [review]
thanks for catching that!

File: src/sparseml/exporters/transforms/convert_quantizable_matmul.py (hunk 2 above)
Comment (corey-nm) [reply src/sparseml/exporters/transforms/convert_quantizable_matmul.py:160]: Just moving it outside the loop so we are only doing it once instead of every time we match. not a bug fix, but an easy optimization

File: src/sparseml/exporters/transforms/gemm_to_matmulinteger_add_cast_mul.py (hunk 1 above)
Comment (corey-nm) [reply src/sparseml/exporters/transforms/gemm_to_matmulinteger_add_cast_mul.py:108]: Any transform that adds nodes will require a call to this

**Review by bfineran — APPROVED** [review]
